Composer/packages/tools/language-servers/language-generation/src/utils.ts

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { TextDocument, Range, Position, DiagnosticSeverity } from 'vscode-languageserver-types';
+import { LGResource, LGParser, DiagnosticSeverity as LGDiagnosticSeverity } from 'botbuilder-lg';
+import get from 'lodash/get';
+
+export interface Template {
+  Name: string;
+  Parameters?: string[];
+  Body: string;
+}
+
+export interface LGDocument {
+  uri: string;
+  content: string;
+  template: Template;
+}
+
+export function getRangeAtPosition(document: TextDocument, position: Position): Range | undefined {
+  let range: Range;
+  const text = document.getText();
+  const line = position.line;
+  const character = position.character;
+  const lineText = text.split('\n')[line];
+  let lastIdx = -1;
+  let start = -1;
+  let end = -1;
+  const regex = /[a-zA-Z0-9_/.]+/;
+  for (let idx = 0; idx < lineText.length; idx++) {
```

Comment:
I think there could be a better implementation here. Use either regex or for-loop, but not both.

For regex, you can also loop by using `g` flag.